packages/playground/src/components/Playground.tsx

```diff
@@ -0,0 +1,410 @@
+      setExtraErrors(extraErrors);
+      setTheme(dataTheme);
+      setArrayFieldTemplate(ArrayFieldTemplate);
+      setObjectFieldTemplate(ObjectFieldTemplate);
+      setShowForm(true);
+    },
+    [onThemeSelected, setShowForm, setUiSchema, setExtraErrors, setTheme, setArrayFieldTemplate, setObjectFieldTemplate]
+  );
+
+  useEffect(() => {
+    const hash = document.location.hash.match(/#(.*)/);
+
+    if (hash && typeof hash[1] === 'string' && hash[1].length > 0) {
+      try {
+        load(JSON.parse(atob(hash[1])));
+      } catch (error) {
+        alert('Unable to load form setup data.');
+        console.error(error);
+      }
+
+      return;
+    }
+
+    // initialize theme
+    onThemeSelected(theme, themes[theme]);
+
+    setShowForm(true);
+  }, [onThemeSelected, load, setShowForm]);
+
+  const onSubthemeSelected = useCallback(
+    (subtheme: any, { stylesheet }: { stylesheet: any }) => {
+      setSubtheme(subtheme);
+      setStylesheet(stylesheet);
+    },
+    [setSubtheme, setStylesheet]
+  );
+
+  const onValidatorSelected = useCallback(
+    (validator: string) => {
+      setValidator(validator);
+    },
+    [setValidator]
+  );
+
+  const handleSetLiveSettings = useCallback(
+    ({ formData }: { formData: any }) => {
+      setLiveSettings(formData);
+    },
+    [setLiveSettings]
+  );
+
+  const onFormDataChange = useCallback(
+    ({ formData }: IChangeEvent<any, RJSFSchema, any>, id?: string) => {
+      if (id) {
+        console.log('Field changed, id: ', id);
+      }
+
+      setFormData(formData);
+      setShareURL(null);
+    },
+    [setFormData, setShareURL]
+  );
+
+  const onFormDataSubmit = useCallback(({ formData }: IChangeEvent<any, RJSFSchema, any>, event: FormEvent<any>) => {
+    console.log('submitted formData', formData);
+    console.log('submit event', event);
+    window.alert('Form submitted');
+  }, []);
+
+  const onShare = useCallback(() => {
+    const {
+      location: { origin, pathname },
+    } = document;
+
+    try {
+      const hash = btoa(
+        JSON.stringify({
+          formData,
+          schema,
+          uiSchema,
+          theme,
+          liveSettings,
+        })
+      );
+
+      setShareURL(`${origin}${pathname}#${hash}`);
+    } catch (error) {
+      setShareURL(null);
+      console.error(error);
+    }
+  }, [setShareURL]);
+
+  const templateProps: Partial<TemplatesType> = {
+    ArrayFieldTemplate,
+    ObjectFieldTemplate,
+  };
+
+  return (
+    <>
+      <div className='page-header'>
+        <h1>react-jsonschema-form</h1>
+        <div className='row'>
+          <div className='col-sm-6'>
+            <Selector onSelected={load} />
+          </div>
+          <div className='col-sm-2'>
+            <Form
+              idPrefix='rjsf_options'
+              schema={liveSettingsSchema as any}
+              formData={liveSettings}
+              validator={localValidator}
+              onChange={handleSetLiveSettings as any}
```

Comment:
I believe the other changes I suggest allows these to not need casts
```suggestion
              schema={liveSettingsSchema}
              formData={liveSettings}
              validator={localValidator}
              onChange={handleSetLiveSettings}
```